source/admin_manual/list_of_events.rst

```diff
@@ -854,579 +487,380 @@ These events apply only for connections using the ``connection API``.
 
-indexer_worker_indexing_finished
-^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Cache
+-----
 
-.. versionadded:: 2.3.15
+.. dovecot_core:event:: mail_cache_decision_changed
+   :added: v2.3.11
 
-Indexer worker process completed an indexing transaction.
+   :field field: Cache field name (e.g. ``imap.body`` or ``hdr.from``).
+   :field last_used: UNIX timestamp of when the field was accessed the last
+     time. This is updated only once per 24 hours.
+   :field reason:
+     Reason why the caching decision changed:
+       * ``add``: no -> temp decision change, because a new field was added to
+         cache.
+       * ``old_mail``: temp -> yes decision change, because a mail older than
+         1 week was accessed.
+       * ``unordered_access``: temp -> yes decision change, because mails
+         weren't accessed in ascending order.
+       * Other values indicate a reason for cache purging, which changes the
+         caching decision yes -> temp.
+   :field uid: IMAP UID number that caused the decision change. This is set
+     only for some reasons, not all.
+   :field old_decision: Old cache decision: ``no``, ``temp``, or ``yes``.
+   :field new_decision: New cache decision: ``no``, ``temp``, or ``yes``.
 
-+---------------------+--------------------------------------------------------+
-| Field               | Description                                            |
-+=====================+========================================================+
-| Inherits from :ref:`event_mailbox`                                           |
-+---------------------+--------------------------------------------------------+
-| message_count       | Number of messages indexed                             |
-+---------------------+--------------------------------------------------------+
-| first_uid           | UID of the first indexed message                       |
-+---------------------+--------------------------------------------------------+
-| last_uid            | UID of the last indexed message                        |
-+---------------------+--------------------------------------------------------+
-| user_cpu_usecs      | Total user CPU spent on the indexing transaction in    |
-|                     | microseconds.                                          |
-+---------------------+--------------------------------------------------------+
+   A field's caching decision changed. The decisions are:
 
+   ========= ================================================================
+   Decision  Description
+   ========= ================================================================
+   ``no``    The field is not cached.
+   ``temp``  The field is cached for 1 week and dropped on the next purge.
+   ``yes``   The field is cached permanently. If the field isn't accessed for
+             30 days it's dropped.
+   ========= ================================================================
 
-Mail cache
-----------
+   .. todo:: Inherit from mail_index
```

Comment:
I guess needs mail_index_common. It doesn't have any fields though.

Reply:
Created an empty field_group named mail_index_common and had all events with this todo inherit from it.